File: iosApp/iosAppTests/UIComponent/DebugMenu/BuildConfigDebug/BuildConfigDebugStateRobot.swift
#if DEBUG
import ComposeApp

final class BuildConfigDebugStateRobot: BaseStateRobot<BuildConfigDebugUiState, BuildConfigDebugEvent> {

    override func defaultState() -> BuildConfigDebugUiState {
        BuildConfigDebugUiState(
            fields: [
                BuildConfigField(name: "appId", value: "us-sampleplatter-mobile-int", group: .identity),
                BuildConfigField(name: "market", value: "us", group: .identity),
                BuildConfigField(name: "baseUrl", value: "https://int-api.sampleplatter.com", group: .urls),
                BuildConfigField(name: "locale", value: "en-US", group: .localization),
            ],
            eventSink: createEventSink(),
        )
    }
}
#endif

File: features/debug-menu/impl/presentation/src/androidDeviceTest/kotlin/com/jkjamies/sampleplatter/features/debugmenu/presentation/BuildConfigDebugStateRobot.kt
package com.jkjamies.sampleplatter.features.debugmenu.presentation

import com.jkjamies.sampleplatter.core.buildconfig.asFields
import com.jkjamies.sampleplatter.core.buildconfig.test.FakeAppBuildConfig
import com.jkjamies.sampleplatter.core.test.StateRobot

class BuildConfigDebugStateRobot : StateRobot<BuildConfigDebugUiState, BuildConfigDebugEvent>() {

    override fun defaultState() = BuildConfigDebugUiState(
        fields = FakeAppBuildConfig().asFields(),
        eventSink = createEventSink(),
    )
}
